portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -20,38 +20,38 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import java.util.ArrayList;
+import com.google.gson.Gson;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  private ArrayList<String> comments = new ArrayList<String>();
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // response.setContentType("text/html;");
-    // response.getWriter().println("<h1>Hello Khadija!</h1>");
-    ArrayList<String> comments = new ArrayList<String>();
-    comments.add("Welcome to my page!");
-    comments.add("Here are own personal comments");
-    comments.add(" Have a great time browsing and finsing out more about me.");
-    String json = convertToJson(comments);
     response.setContentType("application/json:");
+    String json = new Gson().toJson(comments);
```

Comment:
You should use datastore here to load the comments that you've stored. See the "Loading Entities" section of step-5-datastore-walkthrough for an example of how to do this.